src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -1660,6 +1660,60 @@ public string RegularProperty2
                 );
         }
 
+        [Fact, WorkItem(50053, "https://github.com/dotnet/roslyn/issues/50053")]
+        public void PrivatelyImplementingInitOnlyProperty_ReferenceConversion()
+        {
+            string source = @"
+var x = new DerivedType() { SomethingElse = 42 };
+System.Console.Write(x.SomethingElse);
+
+public interface ISomething { int Property { get; init; } }
+public record BaseType : ISomething { int ISomething.Property { get; init; } }
+
+public record DerivedType : BaseType
+{
+    public int SomethingElse
+    {
+        get => ((ISomething)this).Property;
+        init => ((ISomething)this).Property = value;
+    }
+}
+";
+
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            // [ : BaseType::ISomething.set_Property] Cannot change initonly field outside its .ctor.
+            CompileAndVerify(comp, expectedOutput: "42", verify: Verification.Skipped);
+        }
+
+        [Fact, WorkItem(50053, "https://github.com/dotnet/roslyn/issues/50053")]
+        public void PrivatelyImplementingInitOnlyProperty_BoxingConversion()
+        {
+            string source = @"
+var x = new Type() { SomethingElse = 42 };
+
+public interface ISomething { int Property { get; init; } }
+
+public struct Type : ISomething
+{
+    int ISomething.Property { get; init; }
+
+    public int SomethingElse
+    {
+        get => throw null;
+        init => ((ISomething)this).Property = value;
+    }
+}
+";
+
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (13,17): error CS8852: Init-only property or indexer 'ISomething.Property' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
```

Comment:
Would it be reasonable to add a more specific error for this, or do you think that would be more effort than it's worth? The error as it exists is misleading.

Reply:
Looking at the code I don't think it would be a ton of work to add, and I really don't like the error as it exists because it's just wrong.

---
In reply to: [558627354](https://github.com/dotnet/roslyn/pull/50424#discussion_r558627354) [](ancestors = 558627354)